src/main/java/com/codeofus/rent_a_park/controllers/ReservationController.java

```diff
@@ -0,0 +1,65 @@
+package com.codeofus.rent_a_park.controllers;
+
+import com.codeofus.rent_a_park.dtos.ReservationDto;
+import com.codeofus.rent_a_park.errors.BadRequestException;
+import com.codeofus.rent_a_park.mappers.ReservationMapper;
+import com.codeofus.rent_a_park.models.Reservation;
+import com.codeofus.rent_a_park.services.ReservationService;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(path = "/api/v1/reservations")
+public class ReservationController {
+
+    ReservationService reservationService;
+
+    ReservationMapper reservationMapper;
+
+    @GetMapping
+    public List<ReservationDto> getAllReservations(Pageable pageable) {
+        return reservationMapper.reservationToReservationDTO(reservationService.getAllReservations(pageable));
+    }
+
+    @GetMapping("/{id}")
+    public ReservationDto getReservation(@PathVariable int id) throws BadRequestException {
+        Optional<Reservation> reservation = reservationService.getReservation(id);
+        if (reservation.isPresent()) {
+            return reservationMapper.reservationToReservationDTO(reservation.get());
+        } else {
+            throw new BadRequestException("Reservation does not exist", "reservations", "does-not-exist");
+        }
+    }
+
+    @PostMapping
+    public ReservationDto createReservation(@RequestBody ReservationDto reservationDto) throws BadRequestException {
+        if (reservationDto.getId() != null) {
```

Comment:
Why even have id in the class then -> should we handle all possible fields user can send? solution might be to have separate DTO's for update?